### PR TITLE
Story/565 parameterize cli

### LINKED ISSUE #565: ✨ Parameterize CLI E2E
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

It should be able to take the following parameters:

- number of guardians
- quorum
- ballot file
- manifest file

And it should include a way for users to indicate which ballots to spoil

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lprichar**
### Issue
Fixes #565 

### Description
Parameterizes values in the CLI.  More specifically:

- ballot file
- manifest file
- a ballot to spoil

### Testing

1. Try running the new command `run_simple_election.sh`
2. Try running e2e without spoiling anything, like this: `eg e2e --guardian-count=2 --quorum=2 --manifest=data/election_manifest_simple.json --ballots=data/plaintext_ballots_simple.json`

### CODE REVIEW

File: run_simple_election.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+eg e2e --guardian-count=2 --quorum=2 --manifest=data/election_manifest_simple.json --ballots=data/plaintext_ballots_simple.json --spoil-id=25a7111b-4334-425a-87c1-f7a49f42b3a2
```
Comment (lprichar) [inline run_simple_election.sh:1]: A sample script that contains all the values for running e2e for a simple election.  Once we get to #574 this should probably go into docs.

File: src/electionguard_cli/e2e_steps/input_retrieval_step.py [hunk 2]
```diff
@@ -32,9 +32,9 @@ def get_first_value(text: Optional[InternationalizedText]) -> str:
         self.print_value("Contests", len(manifest.contests))
         self.print_value("Ballot Styles", len(manifest.ballot_styles))
 
-    def get_manifest(self) -> Manifest:
+    def get_manifest(self, manifest_file: TextIOWrapper) -> Manifest:
         self.print_header("Retrieving manifest")
-        manifest: Manifest = ElectionFactory().get_simple_manifest_from_file()
+        manifest: Manifest = from_file_wrapper(Manifest, manifest_file)
```
Comment (lprichar) [inline src/electionguard_cli/e2e_steps/input_retrieval_step.py:37]: I considered putting this into ElectionFactory, but it seemed like it was just a passthrough and not adding much value

File: src/electionguard/serialize.py [hunk 3]
```diff
@@ -49,6 +50,13 @@ def to_raw(data: Any) -> Any:
     return json.dumps(data, indent=_indent, default=pydantic_encoder)
 
 
+def from_file_wrapper(type_: Type[_T], file: TextIOWrapper) -> _T:
```
Comment (keithrfung) [inline src/electionguard/serialize.py:53]: This already exists in .serialize in electionguard

File: src/electionguard_cli/commands/e2e_command.py [hunk 4]
```diff
@@ -22,11 +23,39 @@
...
+@click.option(
+    "--manifest",
+    prompt="Manifest file",
+    help="The location of an election manifest.",
+    type=click.File(),
+)
+@click.option(
+    "--ballots",
+    prompt="Ballots file",
+    help="The location of a file that contains plaintext ballots.",
```
Comment (keithrfung) [inline src/electionguard_cli/commands/e2e_command.py:35]: Plaintext ballots should be in a directory. The concatenation was a one off. This truthfully should not be a main path at all if a suggested one at all.

**Review by keithrfung — CHANGES_REQUESTED** [review]
The happy path should use a json file of multiple ballots. I would head for a directory which has multiple files each with a single ballot.

**Comment by keithrfung** [comment]
The conflicts are from the other PR. Be sure to rerun `make auto-lint` the linters are now hooked up correctly and may show new errors after conflict management

File: src/electionguard_cli/cli_models/e2e_inputs.py [hunk 5]
```diff
@@ -14,15 +14,18 @@ def __init__(
         guardians: List[Guardian],
         manifest: Manifest,
         ballots: List[PlaintextBallot],
+        spoil_id: str,
```
Comment (keithrfung) [inline src/electionguard_cli/cli_models/e2e_inputs.py:17]: As a general question, any number of ballots could be spoiled. Shouldn't this just be checking the status of the ballot itself?

**Review by keithrfung — CHANGES_REQUESTED** [review]

File: src/electionguard_cli/e2e_steps/input_retrieval_step.py [hunk 6]
```diff
@@ -15,6 +13,37 @@
 class InputRetrievalStep(E2eStepBase):
     """Responsible for retrieving and displaying user provided inputs."""
 
+    ###########################
+    # Public Mehtods
```
Comment (keithrfung) [inline src/electionguard_cli/e2e_steps/input_retrieval_step.py:17]: *Methods

Comment (keithrfung) [reply src/electionguard_cli/e2e_steps/input_retrieval_step.py:17]: These are frowned upon for the record. 

Just an example:
https://stackoverflow.com/questions/34259939/is-it-pythonic-to-have-header-comments-in-a-python-script

**Review by keithrfung — APPROVED** [review]

File: src/electionguard_cli/cli_models/e2e_inputs.py (hunk 5 above)
Comment (lprichar) [reply src/electionguard_cli/cli_models/e2e_inputs.py:17]: Per our discussion I have created the following tickets: 
- https://github.com/microsoft/electionguard-python/issues/581 for reading in plaintext ballots from directories
- https://github.com/microsoft/electionguard-python/issues/585 for reading in ciphertext ballots from directories
- https://github.com/microsoft/electionguard-python/issues/586 for reading in submitted ballots from a file

Please let me know if I captured the requirements correctly or not.

**Pull request merged**